service/mantle/shopify/ApiServices.xml

```diff
@@ -251,48 +221,25 @@
-                                                              inventory_management:variant.inventory_management,
-                                                              barcode:variant.barcode,
-                                                              weight:variant.weight,
-                                                              weight_unit:variant.weight_unit,
-                                                              sku:variant.sku]))
-                            }else if(featureList.size() == 2){
-                                prodVariants.add(new HashMap([name:variant.name + "/" + variant.option2,
-                                                              option1:variant.option1,
-                                                              option2:variant.option2,
-                                                              price:variant.price,
-                                                              inventory_management:variant.inventory_management,
-                                                              barcode:variant.barcode,
-                                                              weight:variant.weight,
-                                                              weight_unit:variant.weight_unit,
-                                                              sku:variant.sku]))
-                            }else if(featureList.size() == 3){
-                                prodVariants.add(new HashMap([name:variant.name + "/" + variant.option2 + "/" + variant.option3,
-                                                              option1:variant.option1,
-                                                              option2:variant.option2,
-                                                              option3:variant.option3,
-                                                              price:variant.price,
-                                                              inventory_management:variant.inventory_management,
-                                                              barcode: variant.barcode,
-                                                              weight:variant.weight,
-                                                              weight_unit:variant.weight_unit,
-                                                              sku:variant.sku]))
-                            }
+                            variant.inventory_quantity = availableToPromiseTotal
+                            variant.barcode = prodUPC?.idValue
+                            variant.weight = productDimension?.value
+                            variant.weight_unit = uom?.abbreviation
+                            variant.sku = prodSKU?.idValue
+
+                            // we do this so we can identify the variant in the
+                            variant.moquiProductId = assoc.toProductId
+
+                            prodVariants.add(variant)

```

Comment:
This looks a lot cleaner. Good to see that stuff.

Reply:
that was quite a nasty hack, still wondering if there is a way to solve it. The moquiProductId gets thrown away in the very next service.